crates/frontend/src/utils.rs

```diff
@@ -424,3 +425,39 @@ pub fn unwrap_option_or_default_with_error<T>(option: Option<T>, default: T) ->
         default
     })
 }
+
+pub fn get_local_storage<T>(_key: &str) -> Option<T>
+where
+    T: serde::de::DeserializeOwned,
+{
+    cfg_if! {
+        if #[cfg(target_arch = "wasm32")] {
+            web_sys::window()
+                .and_then(|win| win.local_storage().ok())
```

Comment:
```suggestion
                .and_then(|win| win.local_storage().ok().flatten())
```
you can use this to avoid adding the next line `.and_then(|storage| storage)`